File: Application/Services/CustomerService.cs
using AutoMapper;
using DTech.Application.DTOs.request;
using DTech.Application.DTOs.response;
using DTech.Application.DTOs.Response.Admin;
using DTech.Application.DTOs.Response.Admin.Customer;
using DTech.Application.Interfaces;
using DTech.Domain.Entities;
using DTech.Domain.Interfaces;

namespace DTech.Application.Services
{
    public class CustomerService(
        ICustomerRepository customerRepo,
        IOrderRepository orderRepo,
        IProductRepository productRepo,
        IMapper mapper,
        ICloudinaryService cloudinaryService,
        IBackgroundTaskQueue backgroundTaskQueue,
        IEmailService emailService,
        IUnitOfWorkService unitOfWorkService
    ) : ICustomerService
    {
        readonly string folderName = "Pre-thesis/Customer";
        public async Task<CustomerDto> GetCustomer(string customerId)
        {
            var customer = await customerRepo.GetCustomerByIdAsync(customerId);
            if (customer == null)
                return new CustomerDto { Success = false, Message = "Customer not found" };

            var customerDto = mapper.Map<CustomerDto>(customer);
            return customerDto;
        }

        public async Task<MessageResponse> UpdateCustomerProfile(string customerId, UpdateProfileDto model)
        {
            var customer = await customerRepo.GetCustomerByIdAsync(customerId);
            if (customer == null)
                return new MessageResponse { Success = false, Message = "Customer not found" };

            // Check email existed
            var existingEmail = await customerRepo.CheckEmailAsync(model.Email, customer.Id);
            if (existingEmail)
                return new MessageResponse { Success = false, Message = "Email already exists." };


            // Check phone existed
            var existingPhone = await customerRepo.CheckPhoneAsync(model.PhoneNumber, customer.Id);
            if (existingPhone)
                return new MessageResponse { Success = false, Message = "Phone number already exists." };


            // Handle image change
            if (model.ImageUpload != null && model.ImageUpload.Length > 0)
            {
                string imageName = await cloudinaryService.ChangeImageAsync(
                    oldfile: customer.Image ?? string.Empty,
                    newfile: model.ImageUpload,
                    filepath: folderName
                );

                if (string.IsNullOrEmpty(imageName))
                {
                    return new MessageResponse { Success = false, Message = "Image upload failed" };
                }

                model.Image = imageName;
            }

            var newProfile = mapper.Map<ApplicationUser>(model);
            newProfile.Id = customerId;

            var isUpdated = await customerRepo.UpdateCustomerAsync(newProfile);
            if (!isUpdated)
                return new MessageResponse { Success = false, Message = "Failed to update profile" };

            return new MessageResponse { Success = true, Message = "Profile updated successfully" };
        }

        public async Task<MessageResponse> UpdateNewPasswordAsync(string customerId, ChangePasswordDto model)
        {
            var customer = await customerRepo.GetCustomerByIdAsync(customerId);
            if (customer == null)
                return new MessageResponse { Success = false, Message = "Customer not found" };

            var isUpdated = await customerRepo.UpdateCustomerPasswordAsync(customer, model.OldPassword, model.NewPassword);
            if (!isUpdated)
                return new MessageResponse { Success = false, Message = "Failed to change password" };

            return new MessageResponse { Success = true, Message = "Change Password successfully" };
        }

        public async Task<AddressResponse> AddAddressAsync(string customerId, AddAddressDto model)
        {
            var customer = await customerRepo.CheckCustomerAsync(customerId);
            if (!customer)
                return new AddressResponse { Success = false, Message = "Customer not found" };

            var customerAddress = mapper.Map<CustomerAddress>(model);
            customerAddress.CustomerId = customerId;
            var addressId = await customerRepo.AddAddressAsync(customerAddress);
            if (addressId == null)
                return new AddressResponse { Success = false, Message = "Failed to add new address" };

            return new AddressResponse
            {
                Success = true,
                Message = "Add new address successfully",
                AddressId = addressId
            };
        }

        public async Task<MessageResponse> EditAddressAsync(string customerId, EditAddressDto model)
        {
            var customer = await customerRepo.CheckCustomerAsync(customerId);
            if (!customer)
                return new MessageResponse { Success = false, Message = "Customer not found" };

            var customerAddress = mapper.Map<CustomerAddress>(model);
            customerAddress.CustomerId = customerId;
            var isUpdated = await customerRepo.EditAddressAsync(customerAddress);
            if (!isUpdated)
                return new MessageResponse { Success = false, Message = "Failed to edit address" };

            return new MessageResponse { Success = true, Message = "Edit address successfully" };
        }

        public async Task<MessageResponse> DeleteAddressAsync(string customerId, int addressId)
        {
            var isUpdated = await customerRepo.DeleteAddressAsync(customerId, addressId);
            if (!isUpdated)
                return new MessageResponse { Success = false, Message = "Failed to delete address" };

            return new MessageResponse { Success = true, Message = "Delete address successfully" };
        }
        public async Task<MessageResponse> SendContactAsync(ContactDto model)
        {
            var contact = mapper.Map<Feedback>(model);
            var res = await customerRepo.SendContactAsync(contact);
            if (!res)
            {
                return new MessageResponse { Success = false, Message = "Send Contact Fail" };
            }

            if (!string.IsNullOrEmpty(model.Email))
            {
                backgroundTaskQueue.QueueBackgroundWorkItem(async token =>
                {
                    string subject = "Thank you for contacting DTech!";
                    string body = $@"
                        <html>
                            <body style='font-family: Arial, sans-serif; background-color: #f9f9f9; padding: 20px;'>
                                <div style='max-width: 600px; margin: auto; background: white; padding: 20px; border-radius: 10px; box-shadow: 0 2px 5px rgba(0,0,0,0.1);'>
                                    <h2 style='color: #007bff;'>Hi {model.Name},</h2>
                                    <p>Thank you for reaching out to <strong>DTech</strong>!</p>
                                    <p>We've received your message and will get back to you as soon as possible.</p>
                                    <p style='margin-top: 20px;'>Best regards,<br/>The DTech Team</p>
                                </div>
                            </body>
                        </html>";
                    await emailService.SendEmailAsync(model.Email, subject, body);
                });
            }
            return new MessageResponse { Success = true, Message = "Send Contact Successfully" };
        }

        public async Task<AddressResponse> SwitchDefaultAsync(string customerId, int addressId)
        {
            var customer = await customerRepo.CheckCustomerAsync(customerId);
            if (!customer)
                return new AddressResponse { Success = false, Message = "Customer not found" };

            var isUpdated = await customerRepo.SetDefaultAddressAsync(customerId, addressId);
            if (!isUpdated)
                return new AddressResponse { Success = false, Message = "Failed to change default address" };

            return new AddressResponse
            {
                Success = true,
                Message = "Edit address successfully",
                AddressId = addressId
            };
        }

        public async Task<OrderDetailResDto> GetOrderDetailAsync(string customerId, string orderId)
        {
            var customer = await customerRepo.CheckCustomerAsync(customerId);
            if (!customer)
                return new OrderDetailResDto { Success = false, Message = "Customer not found" };

            var order = await orderRepo.GetOrderDetailAsync(customerId, orderId);
            if (order == null)
                return new OrderDetailResDto { Success = false, Message = "Order not found" };

            OrderDetailResDto model = new()
            {
                OrderId = order.OrderId,
                OrderDate = order.OrderDate,
                Name = order.Name,
                NameReceive = order.NameReceive,
                ShippingAddress = order.ShippingAddress,
                Address = order.Address,
                StatusName = order.Status?.Description ?? "Unknown",
                Note = order.Note,
                CostDiscount = order.CostDiscount,
                ShippingCost = order.ShippingCost,
                FinalCost = order.FinalCost,
                Payment = order.Payment == null
                    ? null
                    : new PaymentDto
                    {
                        Status = order.Payment.Status,
                        PaymentMethodName = order.Payment.PaymentMethod!.Description ?? string.Empty,
                    },
                OrderProducts = order.OrderProducts?.Select(op => new OrderProductDto
                {
                    Id = op.Id,
                    Name = op.Product.Name,
                    Photo = op.Product.Photo,
                    Price = op.Price,
                    Quantity = op.Quantity,
                    CostAtPurchase = op.CostAtPurchase,
                }).ToList()
            };

            return model;
        }

        public async Task<OrderDetailResDto> CancelOrderAsync(string customerId, string orderId)
        {
            var customer = await customerRepo.CheckCustomerAsync(customerId);
            if (!customer)
                return new OrderDetailResDto { Success = false, Message = "Customer not found" };

            await unitOfWorkService.BeginTransactionAsync();

            try
            {
                // 1. Update status to Cancelled
                int orderStatusId = 7;
                var orderUpdated = await orderRepo.UpdateStatusAsync(orderId, customerId, orderStatusId);
                if (orderUpdated == null)
                {
                    await unitOfWorkService.RollbackTransactionAsync();
                    return new OrderDetailResDto { Success = false, Message = "Fail to cancel order" };
                }

                // 2. Restore stock for each product
                foreach (var op in orderUpdated.OrderProducts)
                {
                    Console.WriteLine($"[DEBUG REBACK] Restore stock for Product: {op.ProductId}, quantity: {op.Quantity}");
                    bool restored = await productRepo.IncreaseStockAsync(op.ProductId, op.Quantity);
                    if (!restored)
                    {
                        await unitOfWorkService.RollbackTransactionAsync();
                        return new OrderDetailResDto
                        {
                            Success = false,
                            Message = $"Failed to restore stock for Product {op.ProductId}"
                        };
                    }
                }

                await unitOfWorkService.CommitTransactionAsync();

                // 3. Return updated response
                return new OrderDetailResDto
                {
                    Success = true,
                    OrderId = orderUpdated.OrderId,
                    OrderDate = orderUpdated.OrderDate,
                    Name = orderUpdated.Name,
                    NameReceive = orderUpdated.NameReceive,
                    ShippingAddress = orderUpdated.ShippingAddress,
                    Address = orderUpdated.Address,
                    StatusName = orderUpdated.Status?.Description ?? "Unknown",
                    Note = orderUpdated.Note,
                    CostDiscount = orderUpdated.CostDiscount,
                    ShippingCost = orderUpdated.ShippingCost,
                    FinalCost = orderUpdated.FinalCost,
                    Payment = orderUpdated.Payment == null
                        ? null
                        : new PaymentDto
                        {
                            Status = orderUpdated.Payment.Status,
                            PaymentMethodName = orderUpdated.Payment.PaymentMethod!.Description ?? string.Empty,
                        },
                    OrderProducts = orderUpdated.OrderProducts.Select(op => new OrderProductDto
                    {
                        Id = op.Id,
                        Name = op.Product.Name,
                        Photo = op.Product.Photo,
                        Price = op.Product.Price,
                        Quantity = op.Quantity,
                        CostAtPurchase = op.CostAtPurchase,
                    }).ToList()
                };
            }
            catch (Exception ex)
            {
                await unitOfWorkService.RollbackTransactionAsync();
                return new OrderDetailResDto
                {
                    Success = false,
                    Message = "Error: " + ex.Message
                };
            }
        }

        public async Task<MessageResponse> AddProductToWishlistAsync(string customerId, int productId)
        {
            var customer = await customerRepo.CheckCustomerAsync(customerId);
            if (!customer)
                return new MessageResponse { Success = false, Message = "Customer not found" };

            var result = await customerRepo.AddProductToWishlistAsync(customerId, productId);
            if (!result)
                return new MessageResponse { Success = false, Message = "Failed to add product to wishlist" };

            return new MessageResponse { Success = true, Message = "Add product to wishlist successfully" };
        }

        public async Task<IndexResDto<List<WishlistDto>>> GetWishlistAsync(string customerId)
        {
            var customer = await customerRepo.CheckCustomerAsync(customerId);
            if (!customer)
                return new IndexResDto<List<WishlistDto>> {
                    Success = false,
                    Message = "Customer not found"
                };

            var wls = await customerRepo.GetAllWishlistByCustomerIdAync(customerId);

            var wishListDtos = new List<WishlistDto>();
            foreach (var wishlist in wls)
            {
                var product = await productRepo.GetProductByIdAsync(wishlist.ProductId);
                wishListDtos.Add(new WishlistDto
                {
                    WishListId = wishlist.WishListId,
                    ProductId = wishlist.ProductId,
                    Product = product != null ? mapper.Map<ProductDto>(product) : null
                });
            }

            return new IndexResDto<List<WishlistDto>>
            {
                Success = true,
                Data = [.. wishListDtos]
            };
        }

        public async Task<MessageResponse> RemoveProductFromWishlistAsync(string customerId, int productId)
        {
            var customer = await customerRepo.CheckCustomerAsync(customerId);
            if (!customer)
                return new MessageResponse { Success = false, Message = "Customer not found" };

            var isWishlist = await customerRepo.CheckWishlistAsync(customerId, productId);
            if (!isWishlist)
                return new MessageResponse { Success = false, Message = "Failed to remove product from wishlist" };

            var result = await customerRepo.RemoveProductFromWishlistAsync(customerId, productId);
            if (!result)
                return new MessageResponse { Success = false, Message = "Failed to remove product from wishlist" };

            return new MessageResponse { Success = true, Message = "Remove product from wishlist successfully" };
        }

        // For Admin
        public async Task<IndexResDto<List<CustomerIndexDto>>> GetCustomersAsync()
        {
            var customers = await customerRepo.GetAllCustomersAsync();
            if (customers == null || customers.Count == 0)
            {
                return new IndexResDto<List<CustomerIndexDto>>
                {
                    Success = false,
                    Message = "No advertisement found"
                };
            }

            var customerDtos = customers.Select(customer => new CustomerIndexDto
            {
                Id = customer.Id,
                UserName = customer.UserName,
                FullName = customer.FullName,
                Email = customer.Email,
                PhoneNumber = customer.PhoneNumber,
            }).ToList();

            return new IndexResDto<List<CustomerIndexDto>>
            {
                Success = true,
                Data = customerDtos
            };
        }
        public async Task<IndexResDto<CustomerDetailDto>> GetCustomerDetailAsync(string customerId)
        {
            var customer = await customerRepo.GetOnlyCustomerByIdAsync(customerId);
            if (customer == null)
            {
                return new IndexResDto<CustomerDetailDto>
                {
                    Success = false,
                    Message = "Advertisement not found"
                };
            }
            var customerDetail = new CustomerDetailDto
            {
                UserName = customer.UserName,
                FullName = customer.FullName,
                Email = customer.Email,
                PhoneNumber = customer.PhoneNumber,
                Gender = customer.Gender,
                DateOfBirth = customer.DateOfBirth,
                CreateDate = customer.CreateDate,
                CreatedBy = customer.CreatedBy,
                UpdateDate = customer.UpdateDate,
                UpdatedBy = customer.UpdatedBy,
            };

            return new IndexResDto<CustomerDetailDto>
            {
                Success = true,
                Data = customerDetail
            };
        }

    }
}

File: Client/src/services/CustomerService.ts
import axios from "axios";
import { type Customer, type CustomerProfileForm, type ChangePasswordForm } from "../types/Customer";
import { type MessageResponse } from "../types/MessageReponse";
import { type CustomerAddress, type CustomerAddressResponse } from "../types/CustomerAddress";
import type { Contact } from "../types/Contact";
import type { OrderDetailResponse } from "../types/Order";
import type { ServiceResponse } from "../types/Admin";

export const customerService = {
    async getCustomerProfile(token: string): Promise<Customer> {
        try {
            const response = await axios.get<Customer>("/api/profile/me", {
                headers: { Authorization: `Bearer ${token}` },
            });
            return { ...response.data, success: true };
        } catch (err: any) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to fetch customer profile. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async updateCustomerProfile(token: string, data: CustomerProfileForm): Promise<MessageResponse> {
        try {
            const formData = new FormData();
            formData.append("userName", data.userName);
            formData.append("fullName", data.fullName);
            formData.append("email", data.email);
            formData.append("phoneNumber", data.phoneNumber);
            formData.append("gender", data.gender);
            formData.append("dateOfBirth", data.dateOfBirth);
            formData.append("image", data.image);
            if (data.imageUpload) {
                formData.append("imageUpload", data.imageUpload);
            }

            const response = await axios.put<MessageResponse>("/api/profile/update-profile", formData, {
                headers: {
                    Authorization: `Bearer ${token}`,
                },
            });
            return { success: true, message: response.data.message || "Profile updated successfully" };
        } catch (err: any) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to update customer profile. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async changePassword(token: string, data: ChangePasswordForm): Promise<MessageResponse> {
        try {
            const response = await axios.put<MessageResponse>("/api/profile/change-password", data, {
                headers: {
                    Authorization: `Bearer ${token}`,
                },
            });
            return { success: true, message: response.data.message || "Change Password successfully" };
        } catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to change password. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async createAddress(token: string, data: CustomerAddress): Promise<CustomerAddressResponse> {
        try {
            const { addressId, ...dto } = data;
            const response = await axios.post<CustomerAddressResponse>("/api/profile/add-new-address", dto, {
                headers: {
                    Authorization: `Bearer ${token}`,
                },
            });
            return {
                success: true,
                message: response.data.message || "Add new address successfully",
                addressId: response.data.addressId,
            };
        } catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to add new address. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async editAddress(token: string, data: CustomerAddress): Promise<CustomerAddressResponse> {
        try {
            const response = await axios.put<CustomerAddressResponse>("/api/profile/edit-address", data, {
                headers: {
                    Authorization: `Bearer ${token}`,
                },
            });
            return { success: true, message: response.data.message || "Edit address successfully" };
        } catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to edit address. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async deleteAddress(token: string, addressId: number | null): Promise<CustomerAddressResponse> {
        try {
            if (addressId === null) {
                return { success: false, message: "Address not exist" };
            }
            const response = await axios.delete<CustomerAddressResponse>(
                `/api/profile/delete-address/${addressId}`,
                {
                    headers: {
                        Authorization: `Bearer ${token}`,
                    },
                }
            );
            return { success: true, message: response.data.message || "Add new address successfully" };
        } catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to add new address. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async sendContact(data: Contact): Promise<MessageResponse> {
        try {
            const response = await axios.post<MessageResponse>("/api/contact/send-contact", data);
            return { success: true, message: response.data.message || "Send Contact successfully" };
        } catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to send contact. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async setDefaultAddress(token: string, addressId: number): Promise<CustomerAddressResponse> {
        try {
            const response = await axios.put<CustomerAddressResponse>(`/api/profile/switch-default-address/${addressId}`, null, {
                headers: {
                    Authorization: `Bearer ${token}`,
                },
            });
            return {
                success: true,
                message: response.data.message || "Change default successfully" ,
                addressId: response.data.addressId
            };
        } catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to change default address. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async getOrderDetail(token: string, orderId: string): Promise<OrderDetailResponse> {
        try {
            const response = await axios.get<Customer>(`/api/profile/get-order-detail/${orderId}`, {
                headers: { Authorization: `Bearer ${token}` },
            });
            return { ...response.data, success: true, message: "Fetch order detail successfully" };
        } catch (err: any) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to fetch order detail. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async getWishlists<T>(token: string): Promise<ServiceResponse<T[]>> {
        try {
            const response = await axios.get<T[]>(`/api/profile/get-wishlists`, {
                headers: {
                    Authorization: `Bearer ${token}`,
                },
            });
            return {
                success: true,
                data: response.data
            };
        } catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to get wishlists. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async addWishlist(token: string, productId: number): Promise<MessageResponse> {
        try {
            const response = await axios.post<MessageResponse>(`/api/profile/add-wishlist/${productId}`, null, {
                headers: {
                    Authorization: `Bearer ${token}`,
                },
            });
            return {
                success: true,
                message: response.data.message || "Add wishlist successfully",
            };
        } catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to add new wishlist. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },

    async removeWishlist(token: string, productId: number): Promise<MessageResponse> {
        try {
            const response = await axios.delete<MessageResponse>(`/api/profile/remove-wishlist/${productId}`, {
                headers: {
                    Authorization: `Bearer ${token}`,
                },
            });
            return {
                success: true,
                message: response.data.message || "Remove wishlist successfully",
            };
        }catch (err) {
            if (axios.isAxiosError(err)) {
                let message: string = "Failed to add new wishlist. Please try again.";
                if (typeof err.response?.data === "string") {
                    message = err.response.data;
                }
                else if (typeof err.response?.data?.message === "string") {
                    message = err.response.data.message;
                }
                else {
                    message = JSON.stringify(err.response?.data);
                }
                return { success: false, message };
            }
            return { success: false, message: "Unexpected error occurred" };
        }
    },
}